### ISSUE TITLE
[Bug] Using DNS doesn't work

### ISSUE THREAD

**Issue by SchoolGuy** [thkukuk/wireguard-container#5]
When using the container as a peer and utilizing the DNS options, then the container will not work as expected due to the following message in the startup log:

```
Setting container timezone to: Europe/Berlin
Starting setup...
* Interface is set to eth0
* NAT disabled
Start wireguard interface (/etc/wireguard/wg0.conf)
[#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
[#] ip -4 address add 172.30.0.2/24 dev wg0
[#] ip link set mtu 1420 up dev wg0
[#] mount `172.30.0.1 192.168.171.1' /etc/resolv.conf
unshare: unshare(0x20000): Operation not permitted
[#] ip link delete dev wg0
Setting up watches.
Watches established.
```

Wireguard peer config:

```
[Interface]
Address = 172.30.0.2/24
DNS = 172.30.0.1,192.168.171.1
PrivateKey = <my private key>

[Peer]
AllowedIPs = 172.30.0.0/24
Endpoint = <my fqdn>:<my port>
PublicKey = <my public key>
PersistentKeepalive = 25
```

Once I comment out the `DNS` option the container works as expected.

**Comment by thkukuk** [comment]
Yes, `DNS` does not work and doesn't make much sense, since you would need to modify the `/etc/resolv.conf` outside of the container, not inside. Outside is not possible as you have no access to it, inside doesn't seem to be possible with podman.

**Comment by SchoolGuy** [comment]
@thkukuk Isn't it possible to bind mount the file to a desired location and have a script (no idea how that could look like) that would edit the configuration file? The big issue I see is here how to prevent other daemons (like network manager) from overwriting the edited configuration file.

**Comment by thkukuk** [comment]
@SchoolGuy about which file are you speaking and which daemons should overwrite which edited configuration files?

If you mean wg0.conf: **you** as admin are creating it, nobody else is touching it. Nobody else can know that this wireguard tunnel provides a DNS address.

**Comment by SchoolGuy** [comment]
I am referring to `resolv.conf` to get DNS working.

**Comment by thkukuk** [comment]
`resolv.conf` is under control of the container stack, I haven't found a way to modify that inside of the container. You can only do that via podman arguments when starting the container.  And as written, this is a container, so we can only modify things inside the container, which will not help you with your applications running outside of the container and using the wireguard tunnel.

The really better solution here is, as documented: import the wireguard config into your network stack. Using the container on a client was a nice "it's doable, so let's do it" thing, but it doesn't make any sense. Using the network stack for wireguard, too, is much more user friendly and normally doesn't require any additional tools.
